common/src/main/java/org/apache/sedona/common/geometrySerde/UnsafeGeometryBuffer.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sedona.common.geometrySerde;
+
+import org.apache.spark.unsafe.Platform;
```

Comment:
Noticed my build started failing for https://github.com/apache/incubator-sedona/pull/735 because I guess it uses a merged codebase to run off of. The common module is specifically not supposed to have any reference to Spark because:
- It's not versioned by the Scala release
- It's used by Flink as well so this won't even be available

This will need to be removed, either by porting any of the necessary code here directly, or moving this to the core module and making at least the UnsafeGeometryBuffer (or just the whole serializer) a Spark only thing for the time being

Reply:
https://github.com/apache/incubator-sedona/pull/741 was submitted to resolve this issue, where `spark.unsafe.Platform` was replaced by direct references to `sun.misc.Unsafe`.